blocks/to-do/add-and-delete-list-items.js

```diff
@@ -0,0 +1,32 @@
+export default function addListItem(inputValue, list) {
+  // Make sure input is not empty
+  if (!inputValue) {
+    return;
+  }
+
+  // Create elements
+  const listItem = document.createElement('li');
+  const listItemBtn = document.createElement('button');
+
+  // Set attribute
+  listItem.classList.add('to-do__list-item');
+
+  // Change content
+  listItem.textContent = inputValue;
+  listItemBtn.innerHTML = '&#10006;';
+
+  // Append children
+  listItem.appendChild(listItemBtn);
+  list.appendChild(listItem);
+
+  // Delete item function
+  const deleteListItem = (event) => {
+    if (event.target.matches('button')) {
+      const parentListItem = event.target.parentElement;
+      parentListItem.remove();
+    }
+  };
+
+  // Add event listener
+  list.addEventListener('click', deleteListItem);
```

Comment:
We're a bit cleaner now, but we're not fully there yet. In fact, there is a pretty serious memory leak possibility here.

What happens now is every time you add an item (default or manually), you register a new click listener. What makes it even more dangerous is that these event listeners stay with the list even when the items are removed!

Reply:
Note
You can check the currently registered listeners on any element by running this in Chrome: `getEventListeners(document.querySelector('your-element-selector'));`

In this case, `getEventListeners(document.querySelector('.to-do__ordered-list'));` will do the trick.
Check the following:
1. Load the page and run the function in the Chrome console. You'll see 4 click listeners
2. Add an element and run the function again.  You'll see 5 click listeners.
3. Remove 3 elements and run the function again. You'll still see 5 click listeners (but you'll only have 2 items).

Reply:
There are two solutions to this problem.

Solution 1: If you want to add the listener on the list

1. Check if the `list` has an attribute, `data-has-listener` and only add the listener if that attribute is not found
2. When you add the listener, also add the `data-has-listener` attribute to the `list`

This way, your `list` will only get one listener.

Solution 2: If you don't

1. Add the event listener once on the list _outside_ of the `addListItem` function, probably within the `to-do.js`
2. You might need to check the target matcher in this case, but you'll find out if you go with this approach